src/mappings/mappingHandlers.ts

```diff
@@ -128,6 +129,29 @@ export async function handleIbcReceivePacketEvent(cosmosEvent: CosmosEvent): Pro
   await Promise.allSettled([transferRecord.save(), ibcChannel]);
 }
 
+export async function handleBundleInstallMessage(message: CosmosMessage): Promise<void> {
+  const { msg, block, tx } = message;
+
+  if (msg.typeUrl !== '/agoric.swingset.MsgInstallBundle') {
+    logger.warn('message type is not /agoric.swingset.MsgInstallBundle');
+    return;
+  }
+
+  // JSON.stringify converts the object from Uint8Array to readable string
+  const { uncompressedSize, compressedBundle, bundle } = JSON.parse(JSON.stringify(msg.decodedMsg));
+  const bundleRecord = new BundleInstall(
+    tx.hash,
+    BigInt(block.header.height),
+    block.header.time as any,
+    BigInt(uncompressedSize),
+    bundle || '',
+    compressedBundle || '',
+    getAddressFromUint8Array(msg.decodedMsg.submitter),
+  );
+
+  await bundleRecord.save();
+}
+
```

Comment:
I think we should wrap this section of the code in a `try-catch` block to catch and log any errors that may occur.

Reply:
catching the error would result in skipping this message entirely. Throwing an error allows the indexing to stop at that point.
that is our current implementation for all of our functions